### PR TITLE
 Customize symbol keyboards and switch keys 

### PR DESCRIPTION
**Pull request by mirfatif**
1. Always show mirfatif symbol keyboards with mirfatif alphabet keyboards
2. Modified labels for symbol keyboard switch keys for mirfatif keyboards
3. Truncate mirfatif alphabet keyboard name to 3 if is longer than 4 characters

Fit better with my customized QWERTY keyboard and Urdu keyboard.
Collectively they provide a uniform experience: https://github.com/mirfatif/AnySoftKeyboard/blob/master/english_urdu.gif
Doesn't affect other keyboards.

`scripts/ci/ci_check.sh` and `./gradlew testDebugUnitTest` pass.
No tabs in code.

### CODE REVIEW

File: ime/app/src/main/java/com/anysoftkeyboard/keyboards/KeyboardSwitcher.java [hunk 1]
```diff
@@ -294,7 +294,13 @@ private AnyKeyboard getSymbolsKeyboard(int keyboardIndex) {
         ensureKeyboardsAreBuilt();
         AnyKeyboard keyboard = mSymbolsKeyboardsArray[keyboardIndex];
 
-        if (keyboard == null || keyboard.getKeyboardMode() != mKeyboardRowMode) {
+        // if it's one of mirfatif keyboards
+        boolean mirfatifKeyboard = mirfatifKeyboard();
```
Comment (menny) [inline ime/app/src/main/java/com/anysoftkeyboard/keyboards/KeyboardSwitcher.java:298]: what is this type of keyboard?

Comment (mirfatif) [reply ime/app/src/main/java/com/anysoftkeyboard/keyboards/KeyboardSwitcher.java:298]: One of the two keyboards added by me in other PRs: [modified QWERTY Latin keyboard](https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/2409) and [Urdu keyboard](https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/2408).

The changes I made in this PR affect only the two keyboards I'm adding. All other keyboards will keep on functioning as they were.

**Comment by mirfatif** [comment]
@menny is it possible in some way to add customized symbol keyboards and switch keys as independent addons (or as a part of some custom keyboard layout just as a keyboard can override generic rows) without affecting mainstream?

**Comment by menny** [comment]
It is not possible at this time. But it shouldn't be too hard to add this functionality

Do you want to give it a shot?

**Comment by mirfatif** [comment]
> Do you want to give it a shot?

I can try if I'm able to handle this.

File: addons/languages/english/pack/build.gradle [hunk 2]
```diff
@@ -4,3 +4,5 @@ ext.dictionaryInputPossibleCharacters = null//"".toCharArray()
 ext.dictionaryInputAdditionalInnerCharacters = null//"".toCharArray()
 
 apply from: "$rootDir/addons/gradle/language_pack_lib.gradle"
+
+tasks.named('spotlessYaml').get().dependsOn('makeDictionary')
```
Comment (menny) [inline addons/languages/english/pack/build.gradle:8]: This is interesting, why did you add this depend?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Comment (menny) [inline gradle/wrapper/gradle-wrapper.properties:3]: This should be reverted.

File: gradle/root_all_projects_ext.gradle [hunk 4]
```diff
@@ -3,7 +3,7 @@ allprojects {
         androidBuildTools = '34.0.0'
         robolectricVersion = '4.10.3'
 
-        sideBySideNdkVersion = '23.0.7599858'
+        sideBySideNdkVersion = '25.1.8937393'
```
Comment (menny) [inline gradle/root_all_projects_ext.gradle:6]: I will bump the NDK in the next release. Can you revert this?

**Review by menny — CHANGES_REQUESTED** [review]
Thanks for this.
I can not expect this as is: it is too dependant on a specific language.

I suggest the following generalization :
Any keyboard layout may also suggest a list of symbol layouts that will be used when active.
So, the English layout will provide the default, but Urdu will provide the new

**Comment by mirfatif** [comment]
@menny the recent changes aren't meant for the upstream but only for my own fork. I closed this PR before pushing changes to my repo. But you still got notified. I think I should change the branch name.
